src/rgw/rgw_asio_frontend.cc

```diff
@@ -722,7 +722,15 @@ int AsioFrontend::init()
       }
     }
 
-    l.acceptor.set_option(tcp::acceptor::reuse_address(true));
+    // setting option |SO_REUSEPORT| allows running of multiple rgw processes on
+    // the same port. Can read more about the implementation here.
+    // https://web.git.kernel.org/pub/scm/linux/kernel/git/netdev/net-next.git/commit/?id=c617f398edd4db2b8567a28e899a88f8f574798d
+    int one = 1;
+    if (setsockopt(l.acceptor.native_handle(), SOL_SOCKET,
+                   SO_REUSEADDR | SO_REUSEPORT, &one, sizeof(one)) == -1) {
```

Comment:
one downside of this change is that this would no longer fail on _accidental_ reuse of port numbers. while there was a consensus not to require a ceph.conf option for this, it might be best to require opt-in via `so_reuseport` in the [rgw_frontends config](https://docs.ceph.com/en/latest/radosgw/frontends/#options)

Reply:
Sorry your thumbs up mean, we should add soreuse as config for front end ? N then that config will be valid only for beast ?

Reply:
yes i think it's useful to catch accidental reuse of radosgw port numbers. a beast-specific `so_reuseport` option sounds great

Reply:
done, removed the vstart changes and made the resuse_port as the frontend config